pinot-query-planner/src/test/resources/queries/AggregatePlans.json

```diff
@@ -6,14 +6,13 @@
         "sql": "EXPLAIN PLAN FOR SELECT AVG(a.col4) as avg FROM a WHERE a.col3 >= 0 AND a.col2 = 'pink floyd'",
         "output": [
           "Execution Plan",
-          "\nLogicalProject(avg=[/($0, $1)])",
-          "\n  LogicalProject($f0=[CASE(=($1, 0), null:DECIMAL(1000, 0), $0)], $f1=[$1])",
-          "\n    LogicalAggregate(group=[{}], agg#0=[$SUM0($0)], agg#1=[$SUM0($1)])",
-          "\n      LogicalExchange(distribution=[hash])",
-          "\n        LogicalAggregate(group=[{}], agg#0=[$SUM0($0)], agg#1=[COUNT()])",
-          "\n          LogicalProject(col4=[$0], col2=[$1], col3=[$2])",
-          "\n            LogicalFilter(condition=[AND(>=($2, 0), =($1, 'pink floyd'))])",
-          "\n              LogicalTableScan(table=[[a]])",
+          "\nLogicalProject(avg=[/(CASE(=($1, 0), null:DECIMAL(1000, 0), $0), $1)])",
```

Comment:
wonder what made this 2 project merge possible after the rule changes.

Reply:
This is because I am now doing the pruning/merging of redundant operators in the end and using a RuleCollection. (if that's what you meant)

Reply:
ah. gotcha. this is great!